src/wallet/rpcwallet.cpp

```diff
@@ -2626,10 +2628,20 @@ static UniValue loadwallet(const JSONRPCRequest& request)
         }
     }
 
+    bool load_on_startup = false;
+    if (!request.params[1].isNull() && request.params[1].isBool()) {
+        load_on_startup = request.params[1].get_bool();
+    }
+
     std::string error, warning;
     std::shared_ptr<CWallet> const wallet = LoadWallet(*g_rpc_interfaces->chain, location, error, warning);
     if (!wallet) throw JSONRPCError(RPC_WALLET_ERROR, error);
 
+    if (load_on_startup && !AddWalletSetting(wallet->chain(), location.GetName())) {
+        throw JSONRPCError(RPC_MISC_ERROR, "Wallet loaded, but load-on-startup list could not be written, so wallet "
```

Comment:
An error is thrown but wallet remains loaded. Either:
 - make this a warning in the response;
 - move `AddWalletSetting()` before loading - which must be reverted if the actual load fails;
 - unload the wallet before throwing.